app/src/main/java/com/codingblocks/cbonlineapp/adapters/VideosNotesAdapter.kt

```diff
@@ -1,129 +1,51 @@
 package com.codingblocks.cbonlineapp.adapters
 
-import android.content.Context
 import android.view.LayoutInflater
-import android.view.View
 import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
 import androidx.recyclerview.widget.RecyclerView
+import com.codingblocks.cbonlineapp.BR
 import com.codingblocks.cbonlineapp.R
 import com.codingblocks.cbonlineapp.database.models.NotesModel
-import com.codingblocks.cbonlineapp.extensions.retrofitCallback
-import com.codingblocks.cbonlineapp.extensions.secToTime
-import com.codingblocks.cbonlineapp.util.OnItemClickListener
+
 import com.codingblocks.cbonlineapp.viewmodels.VideoPlayerViewModel
-import com.codingblocks.onlineapi.Clients
-import com.codingblocks.onlineapi.models.ContentsId
-import com.codingblocks.onlineapi.models.Notes
-import com.codingblocks.onlineapi.models.RunAttemptsId
-import com.google.android.material.snackbar.Snackbar
-import kotlinx.android.synthetic.main.item_notes.view.bodyTv
-import kotlinx.android.synthetic.main.item_notes.view.contentTitleTv
-import kotlinx.android.synthetic.main.item_notes.view.deleteTv
-import kotlinx.android.synthetic.main.item_notes.view.editTv
-import kotlinx.android.synthetic.main.item_notes.view.timeTv
-import org.jetbrains.anko.design.snackbar
 
 class VideosNotesAdapter(
-    private var notesData: ArrayList<NotesModel>,
-    private val listener: OnItemClickListener,
     private val viewModel: VideoPlayerViewModel
 ) : RecyclerView.Adapter<VideosNotesAdapter.NotesViewHolder>() {
 
-    private lateinit var context: Context
+    var notesData: ArrayList<NotesModel> = ArrayList()
 
     fun setData(notesData: ArrayList<NotesModel>) {
         this.notesData = notesData
         notifyDataSetChanged()
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesViewHolder {
-        context = parent.context
+        val layoutInflater = LayoutInflater.from(parent.context)
+        val binding = DataBindingUtil.inflate<ViewDataBinding>(layoutInflater, R.layout.item_notes, parent, false)
 
-        return NotesViewHolder(
-            LayoutInflater.from(parent.context)
-                .inflate(R.layout.item_notes, parent, false)
-        )
+        return NotesViewHolder(binding)
     }
 
     override fun getItemCount(): Int {
         return notesData.size
     }
 
     override fun onBindViewHolder(holder: NotesViewHolder, position: Int) {
-        holder.bindView(notesData[position], position)
+        holder.bindView(viewModel, position)
```

Comment:
why viewmodel there's no use of viewmodel don't use viewmodels objects inside adapters until there is no alternative for that like SectionDetailsAdapter

Reply:
@aggarwalpulkit596 That is how the architecture works. Please Check this example, https://github.com/alphamu/android-MVVM-DataBinding-RecyclerViewExample. This will clear things out